nihms-native-assembler/src/test/java/org/dataconservancy/nihms/assembler/nihmsnative/NihmsAssemblerIT.java

```diff
@@ -88,9 +89,11 @@ public void testBasicStructure() throws Exception {
         Map<String, File> packageFiles = Arrays.stream(extractedPackageDir.listFiles())
                 .collect(Collectors.toMap((File::getName), Function.identity()));
 
-        packageFiles.keySet().stream()
-                .filter(fileName -> !fileName.equals(manifest.getName()) && !fileName.equals(metadata.getName()))
-                .forEach(fileName -> assertTrue(custodialResourcesMap.containsKey(fileName)));
+        custodialResourcesMap.keySet().stream()
```

Comment:
Can this be executed over the `packageFiles.keySet()` and not the `custodialResourcesMap.keySet()`?  Switching the two changes the semantics of the test in a subtle way.  Iterating over `packageFiles.keySet()` insures that every single file in the package is accounted for, but switching to iterating over `custodialResourcesMap` is essentially re-testing what has already been executed on line 84: the presence of each custodial resource in the package.

Reply:
The problem is that the names in the resource list have had "SUBMITTED-" added to them when there is a collision.  Removing that before testing a file name seemed like a bad idea, given the (slim) chance that a file name genuinely started with that text.  I think I would have to iterate over the contents of the custodial resources key set in an inner loop, add "SUBMITTED-" to each name, and then see if it matched the package file name.  If that sounds OK to you, let me know and I'll do it.

Reply:
So, `packageFiles` would contain a file named `SUBMITTED-blah`, but the `custodialResourcesMap` still has the old name?

Reply:
Yes, that's how it works at the moment.  I think it doesn't actually store the name, but the whole path to the file.  When figuring out what to do for this work, I tried to avoid copying files to new locations to resolve their name conflicts.

Reply:
So the custodial resource map should never include non-custodial content.  With that constraint, then, if a file in the packageFiles map starts with `SUBMITTED-`, you know there was a name conflict.  You could check for the presence of the conflicted version of the name in the custodialResources map.

Come to think of it, the custodial resources check above on line 84 needs to use `SUBMITTED-` version of the filename.  It's succeeding but only because the _real_ metadata file is there as well.

Reply:
Well, that isn't rock-solid, as a submitted file could have been named SUBMITTED-???, but I can do it that way.  Yes, stripping the prefix off so I can look it up in the custodial map would add more certainty, but I'd really like to keep the use of the "SUBMITTED-" string to one file.  I'll think about whether that can be done.

I will look at the resource check loop above wrt your comment.